Add retest.

technomancy opened the pull request:
Fixes #3.

technomancy committed:
Add retest.

Fixes #3.

technomancy commented on src/circleci/test/retest.clj:
```diff
@@ -0,0 +1,23 @@
+      (symbol suite (:name (:attrs test-case))))))
+
+(defn- failed-tests [report-dir]
+  (when-not (.exists (io/file report-dir))
```
Throwing an exception here is not ideal; we really ought to just print a message and abort.

Unfortunately Clojure provides no mechanism whereby you can abort the process with a nonzero exit code when running non-interactively and yet leave the repl running when running interactively, so I don't think we're stuck choosing between two non-ideal options.

technomancy replied:
It occurs to me that we can assume if `-main` is called it won't be interactively, but I should add a public non-main function for repl use.

technomancy commented:
Realized this is more complicated than it looks because `report-dir` is not part of the config map.

@gordonsyme Do you think we should make the directory for the reporter output more declarative instead of making it an argument to `junit/reporter`?

gordonsyme reviewed (approved):
> The consensus seemed to be that if really large numbers of men were sent to storm the mountain, then enough might survive the rocks to take the citadel. This is essentially the basis of all military thinking.

Code lgtm, your concern about JunitReporter and config is reasonable.
I think it'll be awkward to tease them both apart, not sure how you'd get the config into the JunitReporter record.

technomancy committed:
Make reporters read from config.

This is more declarative and allows :test-results-dir to be a config
value so that retest can use it to inspect previous results.

technomancy committed:
Put :test-results-dir in default config.

Update readme re: junit reporter usage.

technomancy committed:
Don't bind *reporters* by default.

technomancy committed:
Fix retest to load failure results properly.

technomancy commented:
I think this is ready now; we merge a default `:test-results-dir` value into the config if none is present and use that from both the junit reporter and the retest function.

technomancy committed:
A few readme tweaks and formatting.

[skip ci]
